packages/contracts/contracts/payment/Vault.sol

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: Apache-2.0
+/// @title Vault
+pragma solidity ^0.8.13;
+
+import {ICartesiDApp} from "@cartesi/rollups/contracts/dapp/ICartesiDApp.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {Pausable} from "@openzeppelin/contracts/security/Pausable.sol";
+
+import {IFinancialProtocol} from "../protocol/IFinancialProtocol.sol";
+import {IVault} from "./IVault.sol";
+
+/// @notice This contract implements an IVault, using an ERC-20 token and a fixed price per unit of time.
+contract Vault is IVault, IFinancialProtocol, Ownable, Pausable {
+    uint256 public immutable price;
+    IERC20 public immutable token;
+    address public immutable payee;
+
+    mapping(ICartesiDApp => uint256) public runway;
+
+    /// @notice Construct a new Vault with the given token as payment method and price
+    /// @param _owner The owner of the vault, who can pause its operation
+    /// @param _token The token to use as payment method
+    /// @param _payee The receiver of the payments to applications managed by this vault
+    /// @param _price The price per second of execution of the application node
+    constructor(address _owner, IERC20 _token, address _payee, uint256 _price) {
+        token = _token;
+        payee = _payee;
+        price = _price;
+
+        // transfer ownership to the given owner
+        _transferOwnership(_owner);
```

Comment:
If you bump `@openzeppelin/contracts` to 5.0.0, then you can make use of the new [`Owanble`](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v5.0.0/contracts/access/Ownable.sol) contract, which now has a constructor which receives the initial owner. This saves some gas during construction. :-)

Reply:
Not moving to v5 now as rollups is still on v4, and it would create a conflict (even though I could import v5 using a npm alias).